getting-started/basicapp/README.md

```diff
@@ -0,0 +1,362 @@
+
+## Packaging in a Docker Container
+The basicapp example can package your application in a Docker container based
+on the [golang image](https://hub.docker.com/_/golang/), with the commands
+```
+docker build -f build/docker/Dockerfile -t basicwebapp-image .
+docker run -itd --rm --name basicwebapp \
+  -p 8080:8080 \
+  --env MESSAGE_SERVICE=mock \
+  basicwebapp-image
+```
+
+This will run the container with the mock messaging service. We will describe
+how to configure the app to connect to Cloud SQL below.
+
+Check that the application is running OK use the same curl commands as above.
+
+Stop the app with the command
+```
+docker stop basicwebapp
+```
+
+Build the Docker container with cloud build:
+```
+gcloud builds submit --config build/cb-docker.yaml .
+```
+
+Run in Docker, test that it works, and stop it
+```
+gcloud auth configure-docker
+PROJECT_ID=[your project]
+docker run -itd --rm -p 8080:8080 \
+  --name basicwebapp \
+  --env MESSAGE_SERVICE=mock \
+  gcr.io/$PROJECT_ID/basicwebapp
+docker ps
+curl -I http://localhost:8080
+docker stop basicwebapp
+```
+
+## Deploy the app to Flex
+To deploy the app to Flex with the database dependency mocked out use the
+command
+```
+gcloud app deploy app-mock.yaml
+```
+
+Check that the app is being served by App Engine with the command
+```
+gcloud app browse
+```
+
+To run the app together with the Cloud SQL database follow steps as in [Using 
+Cloud SQL for 
+MySQL](https://cloud.google.com/appengine/docs/flexible/go/using-cloud-sql) for
+Flex. You can do this by editing the app.yaml file, filling in the text for the
+password. Then redeploy the app using the command
+```
+gcloud app deploy .
+```
+
+You will want to avoid leaving the project id and password in the app
+configuration file, especially if you are adding the file to a code repository.
+Revert the values back to their original symbols by editing app.yaml. To deploy
+with Cloud Build, set the strings as environment variables using the command
+```
+DB_PASSWORD=[user db password]
+gcloud builds submit \
+  --substitutions=_DB_PASSWORD=$DB_PASSWORD \
+  --config build/cb-deploy.yaml .
+```
+
+The [build/cb-deploy.yaml](build/cb-deploy.yaml) file contains a [custom build
+step](https://cloud.google.com/cloud-build/docs/create-custom-build-steps) with
+shell commands to replace the connection string settings in app.yaml with 
+[user defined
+substitutions](https://cloud.google.com/cloud-build/docs/configuring-builds/substitute-variable-values).
+
+Check that the app is being served by App Engine by pointing your browser at
+$PROJECT_ID.appspot.com.
+```
+curl -I $PROJECT_ID.appspot.com
+```
+
+## Testing and Performance Optimization
+The next activities in your development lifecycle probably include integration
+and load testing, profiling, and tracing.
+
+### Integration and Load Testing
+
+Our integration test uses curl to make sure that the application is properly
```

Comment:
Should "curl" be capitalized or have code blocks? Not sure what the convention is. (Applies to the whole file.)

Reply:
Changed to capital (Curl).